### PR TITLE
Add a version of the init method that accepts Context as a parameter

### PR DESCRIPTION
**Pull request by egor-n**

### CODE REVIEW

File: threetenabp/src/main/java/com/jakewharton/threetenabp/AndroidThreeTen.java [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
+  public static void init(Context context) {
     if (initialized.getAndSet(true)) {
       return;
     }
 
     TzdbZoneRulesProvider provider;
     InputStream is = null;
     try {
-      is = application.getAssets().open("org/threeten/bp/TZDB.dat");
+      is = context.getApplicationContext().getAssets().open("org/threeten/bp/TZDB.dat");
```
Comment (JakeWharton) [inline threetenabp/src/main/java/com/jakewharton/threetenabp/AndroidThreeTen.java]: There's no need to call `getApplicationContext()` here.

File: threetenabp/src/main/java/com/jakewharton/threetenabp/AndroidThreeTen.java [hunk 2]
```diff
@@ -11,15 +11,15 @@
 public final class AndroidThreeTen {
   private static final AtomicBoolean initialized = new AtomicBoolean();
 
-  public static void init(Application application) {
```
Comment (JakeWharton) [inline threetenabp/src/main/java/com/jakewharton/threetenabp/AndroidThreeTen.java]: Please keep this as an overload that just call the `Context` version.

**Comment by egor-n** [comment]
I have reverted the `AndroidThreeTenTest` changes, so it goes through both methods.

**Comment by JakeWharton** [comment]
Thanks! I'll make a new release soon.

**Pull request merged**